address.go

```diff
@@ -37,5 +37,33 @@ func NewAddressFromHex(str string) (*Address, error) {
 // ToHex returns the hexadecimal representation of the Address as string.
 // Example: 0x32be343b94f860124dc5fee278fdcbd38c102d88
 func (a *Address) ToHex() string {
-	return "0x" + hex.EncodeToString(a.Address.Bytes())
+	return "0x" + hex.EncodeToString(a.addr.Bytes())
+}
+
+// Addresses is an slice of Address
+type Addresses struct {
+	data []wallet.Address
```

Comment:
Avoid generic names like `data` (unless dealing with something like a stream, where `data` has a meaning). Better use `addrs` or `as` or also `values` or `vs` (then reusable in `BigInts`), which still better conveys the meaning of the field.